lib/src/dartdoc.dart

```diff
@@ -36,14 +37,41 @@ class DartdocFileWriter implements FileWriter {
   @override
   final Set<String> writtenFiles = {};
 
-  DartdocFileWriter(this._outputDir, this.resourceProvider);
+  final int _maxFileCount;
+  final int _maxTotalSize;
+
+  int _fileCount = 0;
+  int _totalSize = 0;
+
+  DartdocFileWriter(
+    this._outputDir,
+    this.resourceProvider, {
+    int? maxFileCount,
+    int? maxTotalSize,
+  })  : _maxFileCount = maxFileCount ?? 0,
+        _maxTotalSize = maxTotalSize ?? 0;
+
+  void _aboutToWrite(String filePath, int length) {
```

Comment:
I think we can give this a name indicating it is checking maximums, like `_validateMaxWriteStats` or something...